### PR TITLE
Upload reward function for console compatibility

### LINKED ISSUE #42: Feature: Copy reward function into S3 model prefix to record which function was used.

### PR DESCRIPTION
**Pull request by jochem725**
This adds a copy of the reward function in the same directory as the training params, which makes the data stored in a bucket compatible with the console import.

### CODE REVIEW

**Review by larsll — CHANGES_REQUESTED** [review]
Can you please also update the upload-model.sh? It should now take all the files from the prefix-path.

**Comment by jochem725** [comment]
@larsll Done :)

File: scripts/upload/upload-model.sh [hunk 1]
```diff
@@ -78,7 +78,7 @@ TARGET_REWARD_FILE_S3_KEY="s3://${TARGET_S3_BUCKET}/${TARGET_S3_PREFIX}/model/re
 TARGET_HYPERPARAM_FILE_S3_KEY="s3://${TARGET_S3_BUCKET}/${TARGET_S3_PREFIX}/ip/hyperparameters.json"
 
 # Check if metadata-files are available
-REWARD_FILE=$(aws $DR_LOCAL_PROFILE_ENDPOINT_URL s3 cp s3://${SOURCE_S3_BUCKET}/${SOURCE_S3_REWARD} ${WORK_DIR} --no-progress | awk '/reward/ {print $4}'| xargs readlink -f 2> /dev/null)
+REWARD_FILE=$(aws $DR_LOCAL_PROFILE_ENDPOINT_URL s3 cp s3://${SOURCE_S3_BUCKET}/${SOURCE_S3_REWARD}/${SOURCE_S3_MODEL_PREFIX}/reward_function.py ${WORK_DIR} --no-progress | awk '/reward/ {print $4}'| xargs readlink -f 2> /dev/null)
```
Comment (larsll) [inline scripts/upload/upload-model.sh:81]: This one is a bug I think; `s3://${SOURCE_S3_BUCKET}/${SOURCE_S3_REWARD}/${SOURCE_S3_MODEL_PREFIX}/reward_function.py` should rather be `s3://${SOURCE_S3_BUCKET}/${SOURCE_S3_MODEL_PREFIX}/reward_function.py`, or?

File: scripts/upload/upload-model.sh [hunk 2]
```diff
@@ -78,7 +78,7 @@ TARGET_REWARD_FILE_S3_KEY="s3://${TARGET_S3_BUCKET}/${TARGET_S3_PREFIX}/model/re
 TARGET_HYPERPARAM_FILE_S3_KEY="s3://${TARGET_S3_BUCKET}/${TARGET_S3_PREFIX}/ip/hyperparameters.json"
```
Comment (larsll) [inline scripts/upload/upload-model.sh:78]: The TARGET_REWARD_FILE_S3_KEY also needs to be adapted to move the file into the root, not model.

File: scripts/upload/upload-model.sh (hunk 1 above)
Comment (larsll) [inline scripts/upload/upload-model.sh]: This should be `s3://${SOURCE_S3_BUCKET}/${SOURCE_S3_MODEL_PREFIX}/reward_function.py` or?

**Review by larsll — CHANGES_REQUESTED** [review]

File: scripts/upload/upload-model.sh (hunk 1 above)
Comment (jochem725) [reply scripts/upload/upload-model.sh]: Whoops, yes I messed this up

File: scripts/upload/upload-model.sh (hunk 2 above)
Comment (jochem725) [reply scripts/upload/upload-model.sh:78]: Ah yes, missed this

**Review by larsll — CHANGES_REQUESTED** [review]
Just did a test here; now there challenge is if you use a dr-upload-model on model created before this PR it will not work. We somehow need to make it backward compatible. Some if-statement around REWARD_FILE to check in the model directory if it does not find in root.

**Comment by jochem725** [comment]
@larsll Yeah that is an issue. I think this should work:

- If reward_function in S3 root:
   - Use this path
- If reward_functio not in S3 root:
   - fall back to old implementation

**Review by larsll — APPROVED** [review]

**Pull request merged**